core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -480,15 +463,15 @@ void CudaInternal::finalize() {
   was_finalized = true;
 
   if (nullptr != m_scratchSpace || nullptr != m_scratchFlags) {
-    using RecordCuda = Kokkos::Impl::SharedAllocationRecord<CudaSpace>;
-    using RecordHost =
-        Kokkos::Impl::SharedAllocationRecord<CudaHostPinnedSpace>;
-
-    RecordCuda::decrement(RecordCuda::get_record(m_scratchFlags));
-    RecordCuda::decrement(RecordCuda::get_record(m_scratchSpace));
-    RecordHost::decrement(RecordHost::get_record(m_scratchUnified));
-    if (m_scratchFunctorSize > 0)
-      RecordCuda::decrement(RecordCuda::get_record(m_scratchFunctor));
+    auto cuda_mem_space = Kokkos::CudaSpace::impl_create(m_cudaDev, m_stream);
+    auto host_mem_space =
+        Kokkos::CudaHostPinnedSpace::impl_create(m_cudaDev, m_stream);
+    cuda_mem_space.deallocate(m_scratchFlags, std::size_t(-1));    // FIXME
```

Comment:
Add a better comment than simply `// FIXME`

Reply:
Updated to pass the actual allocation size